designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ProcessesResources.scala

```diff
@@ -99,10 +100,14 @@ class ProcessesResources(
             complete {
               // To not overload engine, for list of processes we provide statuses that can be cached
               implicit val freshnessPolicy: DataFreshnessPolicy = DataFreshnessPolicy.CanBeCached
-              processRepository
-                .fetchProcessesDetails[Unit](query.toRepositoryQuery)
-                .flatMap(_.map(enrichDetailsWithProcessState[Unit]).sequence)
-                .toBasicProcess
+              (processRepository
+                .fetchProcessesDetails[Unit](
+                  query.toRepositoryQuery
+                ) zip deploymentService.getProcessesInProgress).flatMap { case (processesList, processesInProgress) =>
```

Comment:
Let's move this `deploymentService.getProcessesInProgress` invocation to `DeploymentService.enrich...`. We can enrich all process details instead of doing it for an each process separately. Let's keep as fewer logic as it is possible in the resources

Reply:
After this, the getInProgressActionTypesForAllProcesses become internal so you can change the visibility.